swimlane/core/fields/attachment.py

```diff
@@ -34,6 +34,24 @@ def add(self, filename, stream, content_type=None):
 
         return attachment
 
+    def delete(self, attachment):
```

Comment:
Since there is not issue linked or jira ticket, can some information be added to the PR about why and example usage?
Will there be unit tests for this functionality?

Reply:
This is an important functionality to add.
 

1. The main use case I can think of is if you are syncing records between instances / applications and a parent record attachments have been deleted, you would want it to replicate in the child record(s). 
2. Another use case could be for record ingestion. For example, if you are ingesting an email which has attachments and a record is created but you still have yet to parse that data, if the record has attachments, but you only want attachments of a particular type, then deleting it when parsing the record would be beneficial.